backend/fms_core/services/library.py

```diff
@@ -0,0 +1,118 @@
+from django.core.exceptions import ValidationError
+
+from fms_core.models import LibraryType, Library
+
+from fms_core.services.process import create_process
+from fms_core.services.container import create_container, get_container
+from fms_core.services.sample import transfer_sample, remove_qc_flags
+from fms_core.services.property_value import create_process_properties
+
+from datetime import datetime
+
+
+def get_library_type(name):
+    instrument = None
+    errors = []
+    warnings = []
+    try:
+        library_type = LibraryType.objects.get(name=name)
+    except LibraryType.DoesNotExist as e:
+        errors.append(f"No Experiment Type named {name} could be found.")
+    except LibraryType.MultipleObjectsReturned as e:
+        errors.append(f"More than one Experiment Type was found with the name {name}.")
+
+    return instrument, errors, warnings
+
+def create_library(library_type, library_size, index, platform):
+    library = None
+    errors = []
+    warnings = []
+
+    if not all([library_type, index, platform]):
+        errors.append('Experiment type, Index and Platform are required to create a library.')
+        return library, errors, warnings
+
+    try:
+        library = Library.objects.create(library_type=library_type,
+                                         library_size=library_size,
+                                         index=index,
+                                         platform=platform)
+    except ValidationError as e:
+        errors.append(';'.join(e.messages))
+
+    return library, errors, warnings
+
+
+
+def prepare_library(library_type, library_size, platform, library_date,
```

Comment:
I would match the service signatures to the internal structure instead of the template structure.
The param matching should be done in the importer here the template should be abstracted as much as possible. Maybe we failed so far to do that in other services, but we need to do better. I would have all the information used to define each library in a single iterable object ... inside the source sample info(source_sample, volume_used) and and the library_info(library_type, library_size, platform, index, library_date, library_comment) the other parameter can be separated since they belong with the process (process_properties, protocol, process_comment). At the moment, the date and library_size cannot be set by library they need to be all the same for a process.